### PR TITLE
Make aerosol output frequency variable

### LINKED ISSUE #2072: GOCART AOD files not copied properly
### What is wrong?

The GOCART instantaneous AOD is hard-coded to write out every 6 hours, but the copy routine at the end of the forecast job uses `${FV3_OUTPUT_FH}` to loop through the forecast hours to copy these diagnostic files.

### What should have happened?

These should be consistent. I know the `AERO_HISTORY.rc` file is not dynamic, so perhaps the simplest solution is to force the GOCART POSTDET routine to use 6 hour periods only, but Ideally the AERO_HISTORY file would be configurable.

### What machines are impacted?

All or N/A

### Steps to reproduce

Run the GDAS forecast in ATMA mode 

### Additional information

@andytangborn found the bug

### Do you have a proposed solution?

_No response_

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
The `AERO_HISTORY.rc` file that controls the GOCART output is turned into a template usint `@[ ]` syntax to allow for variable output times. A new "parsing" script is created for GOCART that handles parsing the `AERO_HISTORY.rc` file through `atparse`. Other manip- ulation of these files is also moved to the new parsing script to be consistent with other components.

A new variable, `FHOUT_AERO` is introduced to control the output frequency of the GOCART output. This is initially set to 3 to match other components (this is a change from the current value of 6). Since some aerosol fields are also included as part of the atmos output, it may be desirable to meld this back into `FHOUT` later. However, the atmos output has two frequencies (`FHOUT_HF` and `FHOUT`), a feature not supported by other components.

The RUN lists for aero fcst, aero analysis, and waves is simplified to avoid confusion, since there is no need for them to only be set if the components are on.

Resolves #2072

# Type of change
- [ ] Bug fix (fixes something broken)
- [x] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO

# How has this been tested?
- S2SWA forecast-only on Hercules (manually modifying `AERO_FCST_RUN`)

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

File: parm/config/gfs/config.base
Comment (DavidHuber-NOAA) [inline parm/config/gfs/config.base:181]: I like this!  So much simpler than hunting down the correct location within `APP =~ ...A` if blocks.

**Review by DavidHuber-NOAA — DISMISSED** [review]

File: ush/parsing_namelists_GOCART.sh
Comment (DavidHuber-NOAA) [inline ush/parsing_namelists_GOCART.sh:13]: I think I am missing something.  Where does `inst_aod_freq` actually get written to `AERO_HISTORY.rc`?

File: parm/config/gfs/config.base
Comment (CoryMartin-NOAA) [reply parm/config/gfs/config.base:181]: Same!

**Review by CoryMartin-NOAA — DISMISSED** [review]
I didn't test but this looks good to me

**Review by aerorahul — DISMISSED** [review]
Looks good.
Question:
Not for this PR to address -- how diffferent is the `parm/ufs/gocart/AERO_HISTORY.rc` file in the global-workflow from the one in ufs-weather-model?

File: ush/parsing_namelists_GOCART.sh
Comment (WalterKolczynski-NOAA) [reply ush/parsing_namelists_GOCART.sh:13]: `atparse` does it on L34. This is how the other "parsing" scripts work too.

**Comment by WalterKolczynski-NOAA** [comment]
> Looks good. Question: Not for this PR to address -- how diffferent is the `parm/ufs/gocart/AERO_HISTORY.rc` file in the global-workflow from the one in ufs-weather-model?

@aerorahul Looks like they have already templated the one frequency we use, but not the others. A couple other small changes (also whitespace changes not shown):
```
diff parm/ufs/gocart/AERO_HISTORY.rc sorc/ufs_model.fd/tests/parm/gocart/AERO_HISTORY.rc.IN -b
384c384
<   inst_aod.frequency:  060000,
---
>   inst_aod.frequency:  @[AOD_FRQ],
390a391
>                        'NIEXTTAU'      , 'NI'       , 'AOD_NI',
405,406c406,407
<   tavg_2d_rad.fields:     'CA.bcEXTTAU' , 'CA.bc' ,
< 			  'CA.ocEXTTAU' , 'CA.oc' ,
---
>   tavg_2d_rad.fields:     'CAEXTTAUCA.bc' , 'CA.bc' ,
>                           'CAEXTTAUCA.oc' , 'CA.oc' ,
```

(This is against our develop, not the changes in this PR.)

**Comment by TerrenceMcGuinness-NOAA** [comment]
Hera CI System Wide Failure:

```
Cannot contact Hera-EMC: java.lang.InterruptedException

Run rocotorun.
Gather Rocoto statistics
(C48_ATM_8a18d626 on Hera)
	Total Cycles: 1
	Number Cycles done: 0
	State: RUNNING
Run rocotorun.
/apps/rocoto/1.3.7/bin/rocotorun: fork: retry: Resource temporarily unavailable
/apps/rocoto/1.3.7/bin/rocotorun: fork: retry: Resource temporarily unavailable
/apps/rocoto/1.3.7/bin/rocotorun: fork: retry: Resource temporarily unavailable
/apps/rocoto/1.3.7/bin/rocotorun: fork: retry: Resource temporarily unavailable
/apps/rocoto/1.3.7/bin/rocotorun: fork: Resource temporarily unavailable
```

**Comment by DavidHuber-NOAA** [comment]
**UPDATED**

The `gdas.yaml.j2` archiving yaml picks up on `AERO_ANL_RUN` only, so this needs to be amended to also check the `DO_AERO` flag.  Suggest changing
https://github.com/NOAA-EMC/global-workflow/blob/a42c833abe81df112e80da35843d698677d58bd7/parm/archive/gdas.yaml.j2#L70-L72
to
```jinja
            {% if (AERO_ANL_RUN == "gdas" or AERO_ANL_RUN == "both") and DO_AERO %}
        - "{{ COMIN_CHEM_ANALYSIS | relpath(ROTDIR) }}/{{ head }}aerostat"
            {% endif %}
```

Something similar would also need to be done for `master_gfs.yaml.j2`:
https://github.com/NOAA-EMC/global-workflow/blob/a42c833abe81df112e80da35843d698677d58bd7/parm/archive/master_gfs.yaml.j2#L36-L41

**Comment by WalterKolczynski-NOAA** [comment]
> The `gdas.yaml.j2` archiving yaml picks up on `AERO_ANL_RUN` only, so this needs to be amended to also check the `APP`. Suggest changing
> 
> https://github.com/NOAA-EMC/global-workflow/blob/a42c833abe81df112e80da35843d698677d58bd7/parm/archive/gdas.yaml.j2#L70-L72
> 
> 
> to
> ```
>             {% if (AERO_ANL_RUN == "gdas" or AERO_ANL_RUN == "both") and APP.endswith("A") %}
>         - "{{ COMIN_CHEM_ANALYSIS | relpath(ROTDIR) }}/{{ head }}aerostat"
>             {% endif %}
> ```
> 
> or more explicitly
> 
> ```
>             {% if (AERO_ANL_RUN == "gdas" or AERO_ANL_RUN == "both") and APP in ["ATMA", "ATMWA", "S2SA", "S2SWA"] %}
>         - "{{ COMIN_CHEM_ANALYSIS | relpath(ROTDIR) }}/{{ head }}aerostat"
>             {% endif %}
> ```
> 
> Something similar would also need to be done for `master_gfs.yaml.j2`:
> 
> https://github.com/NOAA-EMC/global-workflow/blob/a42c833abe81df112e80da35843d698677d58bd7/parm/archive/master_gfs.yaml.j2#L36-L41

Why are these not based on `DO_AERO`? That is what they should be based on.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA `DO_AERO` is a much better option instead of using `APP`, though `AERO_ANL_RUN` and `AERO_FCST_RUN` still need to be referenced for the `aerostat` and forecast files.

**Comment by DavidHuber-NOAA** [comment]
I've updated my suggestion above.

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA `DO_AERO` is a much better option instead of using `APP`, though `AERO_ANL_RUN` and `AERO_FCST_RUN` still need to be referenced for the `aerostat` and forecast files.

Presumably, this is only temporary until #2943 is complete. `DO_AERO` should be the only control needed outside of `config.base`.

**Comment by CoryMartin-NOAA** [comment]
@WalterKolczynski-NOAA maybe, there will be a need for a GFS aerosol analysis, but GFS fcst will not have GOCART. I'm not sure if that will be solved by that PR or not.

**Comment by DavidHuber-NOAA** [comment]
> Presumably, this is only temporary until #2943 is complete. `DO_AERO` should be the only control needed outside of `config.base`.

I addressed this in #2943, though two additional flags will now be used: `DO_AERO_ANL` and `DO_AERO_FCST`.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA Should this be reviewed again or do you want to wait until #2943 is merged?

**Comment by WalterKolczynski-NOAA** [comment]
CI Tests set up to run in /lfs/h2/emc/ptmp/walter.kolczynski/PR/PR_2982/RUNTESTS on WCOSS

**Comment by DavidHuber-NOAA** [comment]
@TerrenceMcGuinness-NOAA It looks like this PR finished building on Hera and Hercules, but never started running the tests.  Can you please take a look?

**Comment by TerrenceMcGuinness-NOAA** [comment]
The build processes was interrupted by an unknown internal Java error on the controller.  Restarted both CI runs.

**Comment by DavidHuber-NOAA** [comment]
Looks like `RUNTESTS` needs to be cleaned out.

**Comment by aerorahul** [comment]
> Looks like `RUNTESTS` needs to be cleaned out.

Is this an issue to be resolved in the CI?  As in, when the CI is restarted (label reapplied after a previous failure), should the RUNDIRS be cleaned out as part of driver?  @TerrenceMcGuinness-NOAA what do you think?

**Comment by WalterKolczynski-NOAA** [comment]
Something with the gefs replay is interacting badly with the list of aerosol outputs. Will need to fix it.

File: parm/archive/master_gfs.yaml.j2
Comment (DavidHuber-NOAA) [inline parm/archive/master_gfs.yaml.j2]:
```suggestion
{% if DO_AERO and (AERO_FCST_RUN == "gfs" or AERO_FCST_RUN == "both") %}
```

**Comment by WalterKolczynski-NOAA** [comment]
CI Tests set up to run in /lfs/h2/emc/ptmp/walter.kolczynski/PR/PR_2982/RUNTESTS on WCOSS

**Review by CoryMartin-NOAA — DISMISSED** [review]

File: ush/forecast_postdet.sh
Comment (DavidHuber-NOAA) [inline ush/forecast_postdet.sh:715]: Do you mean `IAU_FHROT`?  Also seeing `IAU_FHROT` recalculated from `config.base` values seems kludge-y/problematic.  Should we open an issue to try and centralize this?

File: ush/forecast_postdet.sh
Comment (DavidHuber-NOAA) [inline ush/forecast_postdet.sh:720]: A comment here would be useful to state why the first forecast hour doesn't have aerosol products and why it depends on IAU.

File: ush/forecast_postdet.sh
Comment (DavidHuber-NOAA) [inline ush/forecast_postdet.sh:730]: Not part of this PR, but was this an problem with HDF5?  I'll see if I can recreate it and open an HDF5 issue.

File: ush/forecast_predet.sh
Comment (DavidHuber-NOAA) [inline ush/forecast_predet.sh:730]:
```suggestion
  # GOCART output times can't be computed here because they may depend on IAU_FHROT in forecast_postdet.sh.
```

File: ush/forecast_postdet.sh
Comment (DavidHuber-NOAA) [reply ush/forecast_postdet.sh:715]:
```suggestion
  # This has to be called during postdet after IAU_FHROT has been set
```

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
The content looks good, just some suggestions on comments and future issues.

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh:715]: Sort of. `fhrot` is what matters. Right now it seems we are just using the `IAU_FHROT` variable for all `fhrot` changes, and then these get converted to `FHROT` in [parsing_model_configure](https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/parsing_model_configure_FV3.sh#L25). Really, we should probably be using `FHROT` here and set it to `IAU_FHROT` (or whatever else) if necessary.

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh:730]: IIRC, it is something in MAPL. I feel like we've been told a couple different times it was fixed but it wasn't.

File: ush/forecast_postdet.sh
Comment (DavidHuber-NOAA) [reply ush/forecast_postdet.sh:715]: OK, thanks for clarifying.

**Review by DavidHuber-NOAA — APPROVED** [review]

**Pull request merged**
